docs/blog/_posts/2018-07-06-ninth-dotty-milestone-release.md

```diff
@@ -0,0 +1,181 @@
+---
+layout: blog-page
+title: Announcing Dotty 0.9.0-RC1
+author: Allan Renucci
+authorImg: /images/allan.jpg
+date: 2018-07-06
+---
+
+Today, we are excited to release Dotty version 0.9.0-RC1. This release serves as a technology
+preview that demonstrates new language features and the compiler supporting them.
+
+Dotty is the project name for technologies that are considered for inclusion in Scala 3. Scala has
+pioneered the fusion of object-oriented and functional programming in a typed setting. Scala 3 will
+be a big step towards realizing the full potential of these ideas. Its main objectives are to
+
+- become more opinionated by promoting programming idioms we found to work well,
+- simplify where possible,
+- eliminate inconsistencies and surprising behaviors,
+- build on strong foundations to ensure the design hangs well together,
+- consolidate language constructs to improve the language’s consistency, safety, ergonomics, and performance.
+
+You can learn more about Dotty on our [website](https://dotty.epfl.ch).
+
+<!--more-->
+
+This is our ninth scheduled release according to our [6-week release schedule](https://dotty.epfl.ch/docs/usage/version-numbers.html).
+The [previous technology preview](https://github.com/lampepfl/dotty/releases/tag/0.8.0-RC1) added
+support for sbt 1, introduced improved unchecked warnings and improved SAM type support.
+
+## What’s new in the 0.9.0-RC1 technology preview?
+
+### Improved REPL [#4680](https://github.com/lampepfl/dotty/pull/4680)
+The REPL now uses [JLine 3](https://github.com/jline/jline3) under the hood which improves on
+many aspects such as auto-completions and multi-line editing. The REPL now also works on Windows!
+
+
+### Documentation support in the IDE [#4461](https://github.com/lampepfl/dotty/pull/4461), [#4648](https://github.com/lampepfl/dotty/pull/4648)
+The Dotty IDE will now display documentation on code hover for symbols that were previously
+compiled by the Dotty compiler. In the future, we plan to let users query the documentation
+in the REPL as well.
+
+
+### Drop requirement that implicit functions must be non-empty [#4549](https://github.com/lampepfl/dotty/pull/4549)
+We decided to remove an arbitrary restriction that implicit and functions must be non-empty.
```

Comment:
We remove the arbitrary restriction that parameters of implicit functions must by non-empty.